lib/ansible/plugins/callback/junit.py

```diff
@@ -208,11 +216,14 @@ def _build_test_case(self, task_data, host_data):
         name = '[%s] %s: %s' % (host_data.name, task_data.play, task_data.name)
         duration = host_data.finish - task_data.start
 
-        if self._task_relative_path:
+        if self._task_relative_path and task_data.path:
             junit_classname = os.path.relpath(task_data.path, self._task_relative_path)
         else:
             junit_classname = task_data.path
 
+        if self._replace_out_of_tree_path is not None and junit_classname.startswith('../'):
+            junit_classname = self._replace_out_of_tree_path + os.path.basename(junit_classname)
```

Comment:
So this would remove the entire directory structure and just leave the configured prefix + leaf filename- theoretically you could have name conflicts that way. Is that going to be a problem?

Reply:
It shouldn't be a problem. It's already possible to have multiple results for the same filename.

The expectation is that only the directory structure is random, not the leaf filename. With that assumption in mind, we can preserve at least some information about where the test result came from.